lib/resources/elasticsearch.rb

```diff
@@ -0,0 +1,140 @@
+    def parse_cluster(content)
+      content['nodes'].map do |node|
+        params_hash = {}
+        params_hash.merge!(parse_node_config(node))
+        params_hash.merge!(parse_node_process(node[1]['process']))
+        params_hash.merge!(parse_node_settings(node[1]['settings']))
+        params_hash.merge!(parse_node_jvm(node[1]['jvm']))
+        params_hash.merge!(parse_node_module_list(node[1]['modules']))
+      end
+    end
+
+    def parse_node_config(node)
+      {
+        'cluster_name'          => content['cluster_name'],
+        'node_id'               => node.first,
+        'node_name'             => node[1]['name'],
+        'host'                  => node[1]['host'],
+        'ip'                    => node[1]['ip'],
+        'version'               => node[1]['version'],
+        'build_hash'            => node[1]['build_hash'],
+        'total_indexing_buffer' => node[1]['total_indexing_buffer'],
+        'roles'                 => node[1]['roles'],
+        'os'                    => node[1]['os'],
+        'transport'             => node[1]['transport'],
+        'http'                  => node[1]['http'],
+        'transport_address'     => { 'ip_addr' => node[1]['transport_address'].split(':')[0], 'port' => node[1]['transport_address'].split(':')[1] },
+      }
+    end
+
+    def parse_node_process(node)
+      {
+        'process'               => node,
+        'mlockall'              => node['mlockall'],
+      }
+    end
+
+    def parse_node_settings(node)
+      {
+        'conf_path'             => node['path']['conf'],
+        'data_path'             => node['path']['data'],
+        'log_path'              => node['path']['logs'],
+        'home_path'             => node['path']['home'],
+        'client_type'           => node['client']['type'],
+        'http_type'             => node['http']['type'],
+        'transport_type'        => node['transport']['type'],
+        'pidfile'               => node['pidfile'],
+      }
+    end
+
+    def parse_node_jvm(node)
+      {
+        'pid'                   => node['pid'],
+        'jvm_version'           => node['version'],
+        'vm_name'               => node['vm_name'],
+        'vm_version'            => node['vm_version'],
+        'vm_vendor'             => node['vm_vendor'],
+        'start_time_in_millis'  => node['start_time_in_millis'],
+        'mem'                   => node['mem'],
+        'gc_collectors'         => node['gc_collectors'],
+        'memory_pools'          => node['memory_pools'],
+        'input_arguments'       => node['input_arguments'],
+        'using_compressed_ordinary_object_pointers' => node['using_compressed_ordinary_object_pointers'],
+
+      }
+    end
+
+    def parse_node_module_list(node)
+      modules_list = {}
+      node.each do |mod|
+        modules_list[mod['name']] = mod
+      end
+      {
+        'module_list' => modules_list.keys,
+      }
+    end
+
+    def to_s
+      'Elasticsearch Environment'
```

Comment:
I think `Environment` is a loaded term... how about `Elasticsearch ClusterInfo`?